File: LeetCode/102. 二叉树的层序遍历.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  const res = []
  const order = (L, R, index) => {
    if (L) {
      if (!res[index]) res[index] = []
      res[index] = [...res[index], L.val]
      order(L.left, L.right, index + 1)
    }
    if (R) {
      if (!res[index]) res[index] = []
      res[index] = [...res[index], R.val]
      order(R.left, R.right, index + 1)
    }
  }
  if (root) {
    res.push([root.val])
    order(root.left, root.right, 1)
  }
  return res
}

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */
var levelOrder = function (root) {
  const res = []
  if (!root) return res
  const queue = []
  queue.push(root)
  while (queue.length) {
    const item = []
    const n = queue.length
    for (let i = 1; i <= n; i++) {
      const cur = queue.shift()
      item.push(cur.val)
      cur.left && queue.push(cur.left)
      cur.right && queue.push(cur.right)
    }
    res.push(item)
  }
  return res
}

File: LeetCode/101-200/102. 二叉树的层序遍历.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func levelOrder(root *TreeNode) [][]int {
	var list []*TreeNode
	var res [][]int
	list = append(list, root)
	count := 0
	for len(list) > 0 {
		n := len(list)
		res = append(res, []int{})
		for i := 0; i < n; i++ {
			node := list[i]
			if node == nil {
				fmt.Println("nil", node, count, res)
				return nil
			}
			res[count] = append(res[count], node.Val)
			if node.Left != nil {
				list = append(list, node.Left)
			}
			if node.Right != nil {
				list = append(list, node.Right)
			}
		}
		count++
		list = list[n:]
	}
	return res
}